gatling-core/src/main/scala/io/gatling/core/funspec/GatlingFunSpec.scala

```diff
@@ -0,0 +1,30 @@
+package io.gatling.core.funspec
+
+import io.gatling.core.Predef._
+import io.gatling.core.action.builder.ActionBuilder
+import io.gatling.core.config.Protocol
+import io.gatling.core.structure.ChainBuilder
+
+import scala.collection.mutable.ListBuffer
+
+abstract class GatlingFunSpec extends Simulation {
+
+  /** Set the protocol configuration used by gatling to execute all specs in this class */
+  def protocolConf: Protocol
+
+  /** Add a spec to be executed */
+  def spec(actionBuilder: ActionBuilder) = specs += actionBuilder
+
+  private[this] val specs = new ListBuffer[ActionBuilder]
+
+  private[this] lazy val testScenario = scenario(this.getClass.getSimpleName)
+    .exec(ChainBuilder(specs.reverse.toList))
+
+  private[core] def setupRegisteredSpecs = {
```

Comment:
The warnings were polluting my shiny warning-less scala 2.11 project :)
Figured i'd fix it while I was at it.